app/graphql/hooks.ts

```diff
@@ -384,6 +437,47 @@ export const useDataCubesObservationsQuery = makeUseQuery<
   DataCubesObservationsData
 >({
   fetch: executeDataCubesObservationsQuery,
+  transform: (data, options) => {
+    const {
+      chartConfig: { conversionUnitsByComponentId },
+    } = options;
+
+    if (
+      !data.data ||
+      !conversionUnitsByComponentId ||
+      Object.keys(conversionUnitsByComponentId).length === 0
+    ) {
+      return data;
+    }
+
+    return {
+      ...data,
+      data: {
+        dataCubesObservations: {
+          ...data.data.dataCubesObservations,
+          data: data.data.dataCubesObservations.data.map((observation) => {
+            const newObservation = { ...observation };
+
+            Object.entries(conversionUnitsByComponentId).forEach(
+              ([componentId, { multiplier }]) => {
+                if (componentId in newObservation) {
+                  const value = newObservation[componentId];
+
+                  if (typeof value === "number") {
+                    newObservation[componentId] = value * multiplier;
+                  } else if (typeof value === "string") {
+                    newObservation[componentId] = Number(value) * multiplier;
+                  }
+                }
+              }
+            );
+
+            return newObservation;
+          }),
+        },
+      },
+    };
+  },
```

Comment:
I wonder how is the data cached and when does the transform happens and what makes it trigger again ? Let's sync IRL ? In react-query, the conversion unit would be part of the uqeyr key but since here we use our own useQuery, I wonder how it works ?